kong/plugins/correlation-id/handler.lua

```diff
@@ -80,7 +80,7 @@ end
 function CorrelationIdHandler:header_filter(conf)
   CorrelationIdHandler.super.header_filter(self)
 
-  if conf.echo_downstream then
+  if conf.echo_downstream and kong.ctx.plugin.correlationid_header_value then
```

Comment:
Would you be able to provide a regression test for this behavior as well? We would need it before merge this. Thanks!

Reply:
Gave a best crack at it by having the test call a route with both the plugin and its echo_downstream enabled and having the request-termination plugin enabled on the same route. I then validate the res is indeed 200 and body is success on the mock call. I suppose without the PR such a test case would throw the error the pdk throws and error out which would reveal a future regression.

Reply:
Oh and I fixed a typo in there for yall 😆 , "genetator" is no more, twas bugging me.